packages/studio-ui/src/web/reducers/flows.ts

```diff
@@ -534,7 +518,7 @@ reducer = reduceReducers(
           id: `skill-${flowRandomId}`,
           type: 'skill-call',
           skill: skillId,
-          name: `${skillId}-${flowRandomId}`,
+          name: payload.nodeName ?? `${skillId}-${flowRandomId}`,
```

Comment:
How could and would `nodeName` be undefined here? Also, what is the outcome if it is? The skill node would have no name?

Reply:
I had to investigate this because I forgot everything over the holidays 😆 

The `requestInsertNewSkill` action is called in two different thunks:

When we create a new skill and insert it, this is called:
https://github.com/botpress/studio/blob/e5bd786a809ffe3cf9c4927022a7be7a3871a975/packages/studio-ui/src/web/views/FlowBuilder/skills/index.jsx#L79-L85
https://github.com/botpress/studio/blob/b957b5b1f505a1443b9e8b30bf018891dbab1339/packages/studio-ui/src/web/actions/index.ts#L398-L401

and when we copy an existing skill and paste it, this is called:

https://github.com/botpress/studio/blob/b957b5b1f505a1443b9e8b30bf018891dbab1339/packages/studio-ui/src/web/actions/index.ts#L212-L219

Creating a new skill doesn't contain a nodeName, so we generate one. Pasting a skill needs to generate the "${sourceNodeName} Copy 1" nodeName, so we pass one.